staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -263,7 +263,7 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 		oldModifyResponse := proxy.ModifyResponse
 		proxy.ModifyResponse = func(response *http.Response) error {
 			code := response.StatusCode
-			if code >= 300 && code <= 399 {
+			if code >= 300 && code <= 399 && len(response.Header.Get("Location")) > 0 {
```

Comment:
What does a client do if it gets a location header that is actually set to empty string?

Reply:
304 according to the test case :)

Reply:
https://pkg.go.dev/net/http#Header.Get
> Get gets the first value associated with the given key. If there are no values associated with the key, Get returns "".

Same as as if it was unset.

Reply:
I would be comfortable assuming all reasonable clients don't pick arbitrary other hostile URLs to redirect to :)

Reply:
Go seems to do the right thing here:

https://github.com/golang/go/blob/7c87012f041948af55d323fc5447230c930e79aa/src/net/http/client.go#L501-L531
https://github.com/golang/go/blob/7c87012f041948af55d323fc5447230c930e79aa/src/net/http/client.go#L631-L637